framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -353,13 +372,54 @@ public static AnnotatedTypeMirror asMemberOf(
                 return atypeFactory.fromElement(elem);
             default:
                 AnnotatedTypeMirror type = asMemberOfImpl(types, atypeFactory, t, elem);
+                type = asMemberOfImpl(types, atypeFactory, t, elem, type);
+
                 if (!ElementUtils.isStatic(elem)) {
                     atypeFactory.postAsMemberOf(type, t, elem);
                 }
                 return type;
         }
     }
 
+    public static AnnotatedTypeMirror asMemberOfPreSubstitution(
```

Comment:
`PreSubstitution` sounds wrong - this is doing a substitution. Once there is javadoc of what the method does we can find a better name.

Reply:
Maybe the asMemberOfImpl is making the name confusing. Actually it does not do a substitution. So I will make asMemberOfImpl more clear.